From 0cddb4c820d39cae8bf6681c644fa353b0c20800 Mon Sep 17 00:00:00 2001
From: Nijin Ashok <nashok@redhat.com>
Date: Mon, 16 Jul 2018 14:42:43 +0530
Subject: [PATCH] [ovirt_node] New plugin for oVirt Node

oVirt Node is a small scaled down version used for hosting virtual
machines. The plugin collects node specific information like
upgrade log, the layer structure etc.

Resolves: #1381

Signed-off-by: Nijin Ashok nashok@redhat.com
Signed-off-by: Bryn M. Reeves bmr@redhat.com
---
 sos/plugins/ovirt_node.py | 41 +++++++++++++++++++++++++++++++++++++++
 1 file changed, 41 insertions(+)
 create mode 100644 sos/plugins/ovirt_node.py

diff --git a/sos/plugins/ovirt_node.py b/sos/plugins/ovirt_node.py
new file mode 100644
index 00000000..ccb5d3c6
--- /dev/null
+++ b/sos/plugins/ovirt_node.py
@@ -0,0 +1,41 @@
+# Copyright (C) 2018 Red Hat, Inc.,
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin
+
+
+class OvirtNode(Plugin, RedHatPlugin):
+    """oVirt Node specific information"""
+
+    packages = (
+        'imgbased',
+        'ovirt-node-ng-nodectl',
+    )
+
+    plugin_name = 'ovirt_node'
+    profiles = ('virt',)
+
+    def setup(self):
+
+        # Add log files
+        self.add_copy_spec([
+            '/var/log/imgbased.log',
+            # Required for node versions < 4.2
+            '/tmp/imgbased.log',
+        ])
+
+        # Collect runtime info
+        self.add_cmd_output([
+            'imgbase layout',
+            'nodectl --machine-readable check',
+            'nodectl info',
+        ])
+
+
+# vim: expandtab tabstop=4 shiftwidth=4
-- 
2.17.2

